### PR TITLE
Feature/page wide search

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Page Wide Search

### CODE REVIEW

**Comment by huayunh** [comment]
Hmm, if you first go to a page with white background, say, http://localhost:4200/forms-and-validation/in-a-list, and then you navigate to page-wide search, you would see that the background stays white. But if you go to the Page-Wide Search page directly from the landing page, it renders OK.

image

**Review by huayunh — COMMENTED** [review]
image
This should not be an icon button. 

image
Can we use the standard / underlined form field variant for this guy?

Functionality-wise everything looks nice

**Comment by emclaug2** [comment]
> Hmm, if you first go to a page with white background, say, http://localhost:4200/forms-and-validation/in-a-list, and then you navigate to page-wide search, you would see that the background stays white. But if you go to the Page-Wide Search page directly from the landing page, it renders OK.

We've got another page's styles bleeding into this one; I'll figure out where it's coming from & fix it.

**Comment by emclaug2** [comment]
> We've got another page's styles bleeding into this one; I'll figure out where it's coming from & fix it.

There's quite a bit of style bleeds from the form examples which will require that I edit a large amount of files.  Sounds to me a like a separate PR just to fix the style bleeding.

**Comment by emclaug2** [comment]
#138  opened to fixed style bleeds.

**Comment by emclaug2** [comment]
> Can we use the standard / underlined form field variant for this guy?

Should be good now & ready for another round of review @huayunh

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -56,6 +57,11 @@ export const ROUTES: DrawerItem[] = [
                 path: 'search',
```
Comment (huayunh) [inline src/app/app-routing.module.ts:57]: Let's call this "global search" to differentiate

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        <div class="mat-title" [style.marginLeft.px]="isSmall ? 24 : 0">Page Search</div>
+        <pxb-spacer></pxb-spacer>
+
+        <div pxb-actions class="appbar-actions">
+            <button mat-icon-button>
+                <mat-icon style="margin: 0 8px">help</mat-icon>
+            </button>
+            <button mat-icon-button>
+                <mat-icon style="margin: 0 8px" [matBadge]="88" matBadgeColor="warn">notifications</mat-icon>
+            </button>
```
Comment (huayunh) [inline src/app/pages/app-bar/page-wide-search/page-wide-search.component.html:15]: `0 4px` margin works better for these icons

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<mat-toolbar data-cy="pxb-toolbar" color="primary" class="main-header mat-elevation-z4 t" [class.hidden]="toggleSearch">
```
Comment (huayunh) [inline src/app/pages/app-bar/page-wide-search/page-wide-search.component.html:1]: what does this `t` class do?

Comment (huayunh) [reply src/app/pages/app-bar/page-wide-search/page-wide-search.component.html:1]: And it looks like this shadow has gone missing
image

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+</mat-toolbar>
+<mat-divider *ngIf="isSmall"></mat-divider>
+
+<div class="content-body-container" [class.small]="isSmall">
+    <div class="content-body" [class.small]="isSmall">
+        <div *ngIf="!isSmall" class="desktop-search">
+            <pxb-spacer></pxb-spacer>
+            <mat-form-field appearance="standard" style="width: 160px">
+                <mat-icon matPrefix class="desktop-search-icon">search</mat-icon>
+                <input matInput type="text" placeholder="Search Asset" [(ngModel)]="searchText" />
```
Comment (huayunh) [inline src/app/pages/app-bar/page-wide-search/page-wide-search.component.html:53]: Make your desktop and mobile text match each other

```suggestion
                <input matInput type="text" placeholder="Search Assets..." [(ngModel)]="searchText" />
```

**Review by huayunh — APPROVED** [review]

**Comment by jeffvg** [comment]
did we want to include 'X' to clear search like what was done with the react example? @huayunh 

image

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html (hunk 2 above)
Comment (emclaug2) [reply src/app/pages/app-bar/page-wide-search/page-wide-search.component.html:15]: Ok you got it.

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html (hunk 3 above)
Comment (emclaug2) [reply src/app/pages/app-bar/page-wide-search/page-wide-search.component.html]: I've re-added the elevation here.

File: src/app/pages/app-bar/page-wide-search/page-wide-search.component.html (hunk 4 above)
Comment (emclaug2) [reply src/app/pages/app-bar/page-wide-search/page-wide-search.component.html]: Fixed 👍

**Comment by huayunh** [comment]
when the text field isn't empty, there shall be an X, I think.

**Comment by emclaug2** [comment]
> when the text field isn't empty, there shall be an X, I think.

Yep and this is how this example is handling it; whenever the X is hit the search is cleared.

**Pull request merged**